libs/platform/user/kernel_um.cpp

```diff
@@ -518,6 +530,10 @@ IoGetFileObjectGenericMapping() { return &_mapping; }
 NTSTATUS
 RtlCreateAcl(_Out_ PACL Acl, unsigned long AclLength, unsigned long AclRevision)
 {
+    if (ebpf_fault_injection_inject_fault()) {
+        return STATUS_INVALID_PARAMETER;
```

Comment:
One should never return INVALID_PARAMETER due to fault injection.
INVALID_PARAMETER means there's a bug in the caller.
Here I don't think there's a valid case for using fault injection in this function at all, since from MSDN it seems the only time it can fail is a caller bug.

Reply:
As per my understanding, the goal for fault injection is to ensure the callers handle the return error case so the application does not crash or leak in production (and to help not to regress in maintenance).

In this case,  I can return  'STATUS_BUFFER_TOO_SMALL' instead. Or, do you want to remove fault injection from this API?

Reply:
There are only 3 return status, as per MSN doc.

STATUS_SUCCESS | The ACL was successfully created and initialized.
STATUS_BUFFER_TOO_SMALL | The new ACL does not fit into the buffer at Acl. A larger ACL buffer is required.
STATUS_INVALID_PARAMETER  | The specified revision is not current, or the value of AclLength is too large.

Then, I can remove fault injection from this API.

Reply:
Removed fault injection from this API.